lib/makeLODF.m

```diff
@@ -1,23 +1,51 @@
-function LODF = makeLODF(branch, PTDF);
+function LODF = makeLODF(branch, PTDF, mask_bridge);
 %MAKELODF   Builds the line outage distribution factor matrix.
-%   LODF = MAKELODF(BRANCH, PTDF) returns the DC line outage
-%   distribution factor matrix for a given PTDF. The matrix is nbr x nbr,
-%   where nbr is the number of branches.
+%   LODF = MAKELODF(MPC, PTDF)
+%   LODF = MAKELODF(BRANCH, PTDF)
+%   LODF = MAKELODF(MPC, PTDF, MASK_BRIDGE)
+% 
+%   Returns the DC line outage distribution factor matrix for a given PTDF.
+%   The matrix is nbr x nbr, where nbr is the number of branches.
 %
 %   Example:
-%       H = makePTDF(baseMVA, bus, branch);
+%       H = makePTDF(mpc);
 %       LODF = makeLODF(branch, H);
+%       LODF = makeLODF(mpc, H);
 %
-%   See also MAKEPTDF.
+%       % mask bridge branches in LODF
+%       makeLODF(mpc, H, 1);
+%
+%   See also MAKEPTDF, FIND_BRIDGES.
 
 %   MATPOWER
 %   Copyright (c) 2008-2016, Power Systems Engineering Research Center (PSERC)
-%   by Ray Zimmerman, PSERC Cornell
+%   by Ray Zimmerman, PSERC Cornell,
+%   and Liangyu Zhang, Zhejiang University
+%
+%   Modified by Liangyu Zhang
+%   2023.05.19 Mask corresponding columns of network bridge branches
+
 %
 %   This file is part of MATPOWER.
 %   Covered by the 3-clause BSD License (see LICENSE file for details).
 %   See https://matpower.org for more info.
 
+if isstruct(branch)
+    mpc = branch;
+    branch = mpc.branch;
+end
+
+if nargin < 3
+    mask_bridge = 0;
+else %% nargin >= 3
+    if ~exist('mpc', 'var')
+        error('MPC is required when mask_bridge is enabled.');
+    end
+end
+if ~exist('mask_bridge', 'var')
+    mask_bridge = 0;
+end
```

Comment:
Isn't case already included in lines 38-39?

Reply:
You're right, I've removed them.